### DISCUSSION TITLE
How to load relations dynamically.

### DISCUSSION THREAD

**Discussion by xiaomlove** [filamentphp/filament#3947] [Help] [answered]
The static method getActions() returns an array，how to decide which relationship to show based on the ownerRecord. Or there can be methods in RelationManager to control itself visibility base on ownerRecord.

**Comment by danharrin** [comment, marked as answer]
https://filamentphp.com/docs/2.x/admin/resources/relation-managers#conditional-visibility

**Comment by xiaomlove** [comment]
Thanks so much! @danharrin 
I have another problem: I want to control the relation tab text base on ownerRecord, How can I do that? The static function `getModelLabel()` doesn't have paramter of ownerRecord.

**Reply by danharrin** [reply]
I don't think that's possible atm, sorry

**Reply by xiaomlove** [reply]
> I don't think that's possible atm, sorry

I now get the record data from the route to get the id or request parameters, which also serves the purpose, so hopefully there will be more simple ways to do this in the future.
